packages/bot-web-ui/src/stores/journal-store.js

```diff
@@ -76,14 +80,20 @@ export default class JournalStore {
         const unique_id = Blockly.utils.genUid();
 
         this.unfiltered_messages.unshift({ date, time, message, message_type, className, unique_id, extra });
+        localStorage.setItem(journal_storage_key, JSON.stringify(this.unfiltered_messages));
     }
 
     @computed
     get filtered_messages() {
-        // filter messages based on filtered-checkbox
-        return this.unfiltered_messages.filter(
-            message =>
-                this.journal_filters.length && this.journal_filters.some(filter => message.message_type === filter)
+        return (
+            this.unfiltered_messages
+                .concat(JSON.parse(localStorage.getItem(journal_storage_key)))
+                // filter messages based on filtered-checkbox
```

Comment:
Could we instead restore these messages when a user is logging in once? i.e. `onClientInit` (see `base_store.js`), there is no need for us to keep querying the local storage whenever `this.unfiltered_messages` changes because, if it's initialised once, it already has the same values as in memory.